library/rmc/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+#![feature(rustc_attrs)] // Used for rustc_diagnostic_item.
+
+/// Creates an assumption that will be valid after this statement run. Note that the assumption
+/// will only be applied for paths that follow the assumption. If the assumption doesn't hold, the
+/// program will exit successfully.
+///
+/// # Example:
+///
+/// The code snippet below should never panic.
+///
+/// ```rust
+/// let i : i32 = rmc::nondet();
+/// rmc::assume(i > 10);
+/// if i < 0 {
+///   panic!("This will never panic");
+/// }
+/// ```
+///
+/// The following code may panic though:
+///
+/// ```rust
+/// let i : i32 = rmc::nondet();
+/// assert!(i < 0, "This may panic and verification should fail.");
+/// rmc::assume(i > 10);
+/// ```
+#[inline(never)]
+#[rustc_diagnostic_item = "RmcAssume"]
+pub fn assume(_cond: bool) {}
```

Comment:
For a better docs and IDE experience, it would be preferable to do something like:

```rust
pub fn assume(cond: bool) {
    let _ = cond;
}
```

Same goes for the other arguments.

Reply:
Sure. I'll fix this next time I push some updates to this crate. Thanks for the suggestion.